storybookjs/storybook
# Rethrowing errors caught while loading stories

Issue:

Fixes (partially) https://github.com/storybookjs/storybook/issues/13151

## What I did

Ensured that errors encountered while loading stories are re-thrown without preventing subsequent stories from loading.

@luke-j added the try / catch in https://github.com/storybookjs/storybook/pull/10974, in his description he says, "Re-throwing the error is not an option as that is also swallowed." However, re-throwing asynchronously, in a `setTimeout`, prevents it from getting swallowed.

## How to test

- Is this testable with Jest or Chromatic screenshots? Possibly, though I'm not sure how to go about mocking a module load error. I'm open to suggestions.
- Does this need a new example in the kitchen sink apps? No
- Does this need an update to the documentation? No

If your answer is yes to any of these, please make sure to include it in your PR.
